src/transport/websocket.ts

```diff
@@ -507,12 +506,12 @@ export default class WebSocketTransport implements SubscriptionTransport {
   private onCloseMessage(closeMessage: Message) {
     const [statusCode, headers, body] = closeMessage;
     if (typeof statusCode !== 'number') {
-      return this.close(new Error('Close message: Invalid EOS Status Code'));
+      return this.close(new ProtocolError('Close message: Invalid EOS Status Code'));
```

Comment:
if `close` is `void` then it's confusing to return it. Should surely be `this.close(...); return`?